### DISCUSSION TITLE
ReactionFromSmarts does not produce products for a rearrangement reaction

### DISCUSSION THREAD

**Discussion by cottonkiet** [rdkit/rdkit#5123] [Q&A]
Hi

I am using the following reaction:
`[#6:12]-[#8:11]-[#6:1]-1=[#6:9]-2-[#6:8]=[#6:7]-[#6:6]=[#6:5]-[#6:10]-2=[#6:4]-[#6:3]=[#6H1:2]-1>>[#6:12]-[#6:2]-1=[#6:3]-[#6:4]=[#6:10]-2-[#6:5]=[#6:6]-[#6:7]=[#6:8]-[#6:9]-2=[#6:1]-1-[#8:11]`
with one reactant: 
`CCN(CC)C(=O)C(C)Oc1cccc2ccccc12`
but produce no product
In theory, it should produce a product
`CCN(CC)C(=O)C(C)c1ccc2ccccc2c1O`

To reproduce

```
from rdkit import Chem
from rdkit.Chem import AllChem
react = Chem.MolFromSmiles('CCN(CC)C(=O)C(C)Oc1cccc2ccccc12')
display(react)
scheme = '[#6:12]-[#8:11]-[#6:1]-1=[#6:9]-2-[#6:8]=[#6:7]-[#6:6]=[#6:5]-[#6:10]-2=[#6:4]-[#6:3]=[#6H1:2]-1>>[#6:12]-[#6:2]-1=[#6:3]-[#6:4]=[#6:10]-2-[#6:5]=[#6:6]-[#6:7]=[#6:8]-[#6:9]-2=[#6:1]-1-[#8:11]'
rxn = AllChem.ReactionFromSmarts(scheme)
display(rxn)
prod = rxn.RunReactants([react])
len(prod)
display(prod)

```
Configuration (please complete the following information):

RDKit version: 2021.09.5
OS: Windows
Python version (if relevant): 3.9
Are you using conda? Yes
If you are using conda, which channel did you install the rdkit from? default

**Comment by bp-kelley** [comment]
The issue you are having is that the reaction is written in Kekule form while the smiles are in aromatic form.  If you switch the reaction to aromatic bonds:

```
scheme = '[#6:12]-[#8:11]-[#6:1]:1:[#6:9]:2:[#6:8]:[#6:7]:[#6:6]:[#6:5]:[#6:10]:2:[#6:4]:[#6:3]:[#6H1:2]:1>>[#6:12]-[#6:2]-1=[#6:3]-[#6:4]=[#6:10]-2-[#6:5]=[#6:6]-[#6:7]=[#6:8]-[#6:9]-2=[#6:1]-1-[#8:11]'
```

![image](https://user-images.githubusercontent.com/9888416/159696800-8aaad404-e985-460a-86e7-cf358effd905.png)

The product appears as desired:

![image](https://user-images.githubusercontent.com/9888416/159696727-be74e2b4-0878-4329-a1c7-93ccad4f6042.png)

Note that kekulizing the smiles may or may not work depending on how bond orders are assigned, so aromatic reaction smarts are a better bet in general.

**Reply by cottonkiet** [reply]
Thanks so much! That solves my problem. I just checked: if the smarts is aromatic, it will work no matter whether the smiles is kekule or aromatic; but if the smarts is kekule, it hardly works. 
May I ask what code do you use to transfer the kekule smarts to aromatic smarts? Thanks a lot!

**Comment by bp-kelley** [comment]
I did it by hand.  Sometimes AllChem.SanitizeRxn works as it tries to
aromatize if there is enough information, but here there wasn't.

**Reply by cottonkiet** [reply]
Thanks a lot! 
a similar question arises when I am doing another reaction. This time, no kekule form is involved, but still I am getting no result.  It seems that the problems lays on matching the N-C(=O)-N part of the smarts, though. Thanks a lot for your help. 
```
react = Chem.MolFromSmiles('COc1ccc(NC(=O)N(C)C)cc1Cl')
scheme = '[#6;a:1]-[#7:2]-[#6:3](=[O:4])-[#7:5][C:6]([#1,#6:7])([#1,#6:9])[#1,#6:8]>>[#6;a:1]-[#7:2]-[#6:3](-[#7:5])=[O:4]'
```

**Reply by greglandrum** [reply]
This one is dying because the molecule you provide can't match unless you add explicit Hs to it.
By default molecules in the RDKit do not have explicit Hs on them, so you need to be careful about query patterns which include `#1`
In your case you don't actually need the reactant atoms with mapping numbers 6-9 in your reaction anyway, so you could replace the reactant with something like this (not tested... you may need to tweak it):
```
scheme = '[#6;a:1]-[#7:2]-[#6:3](=[O:4])-[#7:5]-[C;!$(C(-[#7])-[!#1;!#6])]>>[#6;a:1]-[#7:2]-[#6:3](-[#7:5])=[O:4]'
```

**Reply by cottonkiet** [reply]
Thanks so much! I checked the code and it works! It's much more concise than the recursive smarts I came up with. 
`scheme = '[#6;a:1]-[#7:2]-[#6:3](=[O:4])-[#7:5][C;$([CH3]),$([CH2]C),$([CH1](C)C),$(C(C)(C)C):6]>>[#6;a:1]-[#7:2]-[#6:3](-[#7:5])=[O:4]'`
Thanks a lot!
